packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -750,6 +750,14 @@ class AnimationController extends Animation<double>
     return result;
   }
 
+  /// Update the simulation without restarting the animation.
+  ///
+  /// The current simulation will be replaced with the provided [Simulation].
+  void updateSimulation(Simulation simulation){
+    assert(isAnimating);
```

Comment:
The docs should mention that this may only be called in this state.

Reply:
this assert is not necessary. so i remove it.